lib/db.js

```diff
@@ -349,44 +367,32 @@ class LimitDBRedis extends EventEmitter {
     }
 
     const count = params.count || 50;
-    const transform = this.buildStatusResponse(bucket, prefixLength);
+    const transform = this.buildStatusResponse(params.type, bucket, prefixLength);
 
-    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
-    const errors = [];
-    async.map(nodes, (node, cb) => {
-      if (!node.bucketStatus) {
-        node.defineCommand('bucketStatus', {
-          numberOfKeys: 0,
-          lua: STATUS_LUA
-        });
-      }
-      node.bucketStatus(prefix, count, (err, results) => {
-        if (err) {
-          errors.push(err);
-          this.emit('error', err);
-        }
-        return cb(null, results || []);
-      });
-    }, (err, results) => {
-      if (err) {
-        // Should not happen
-        return callback(err);
-      }
+    const matches = [];
+    let errors = [];
+    const stream = this.scanStream({ prefix });
 
-      const items = [];
-      for (let i = 0; i < results.length; i++) {
-        for (let j = 0; j < results[i].length; j++) {
-          const match = results[i][j];
-          items.push(transform(match[2], parseInt(match[0], 10), parseInt(match[1], 10)));
-          if (items.length === count) {
-            break;
-          }
-        }
-        if (items.length === count) {
+    stream.on('error', (err) => {
+      errors.push(err);
+    });
+
+    stream.on('data', (keys) => {
+      for (let i = 0; i < keys.length; i++) {
+        matches.push(keys[i]);
+        if (matches.length >= count) {
           break;
         }
       }
-      return callback(null, { items, errors });
+      if (matches.length >= count) {
+        stream.end();
+      }
+    });
+
+    stream.on('end', () => {
+      async.map(matches, transform, (err, items) => {
```

Comment:
How much parallelism do we have here? Is there any limit or it is full series?

Reply:
Why not to use a stream transform and transform on the fly instead of waiting till end?

Reply:
What happens with the errors while transforming?

Reply:
Swapped.